src/oemof/solph/flows/_non_convex_flow_block.py

```diff
@@ -743,17 +743,11 @@ def _positive_gradient_flow_constraint(_):
                             m.flow[
                                 i,
                                 o,
-                                m.TIMEINDEX[index][0],
-                                m.TIMEINDEX[index][1],
+                                m.TIMESTEPS[index],
```

Comment:
Kind of not directly related, but why does line `740` iterate over a numeric `index` instead of `t` as is done everywhere else? I guess there is a good reason for it...

Reply:
To my understanding, this is because of the `positive_gradient_constr` needing the `TIMEINDEX` (including periods) to be allowed to consider investment (periods) in the ramping. (@jokochems might now better.) There is probably a more elegant way to do this, but I did not want to touch this now.

Reply:
Maybe also something to be discussed in a separate issue to not delay the progress too much

Reply:
If it is really the periods, I did something wrong, as `m.TIMESTEPS[p]` does not make sense.